Explorer/Assets/Scripts/SceneRunner/SceneFactory.cs

```diff
@@ -140,7 +141,15 @@ private async UniTask<ISceneFacade> CreateSceneAsync(ISceneData sceneData, IPart
                 sceneData.TryGetMainScriptUrl(out sceneCodeUrl);
             }
 
-            SceneRuntimeImpl sceneRuntime = await sceneRuntimeFactory.CreateByPathAsync(sceneCodeUrl, instancePoolsProvider, sceneData.SceneShortInfo, ct, SceneRuntimeFactory.InstantiationBehavior.SwitchToThreadPool);
+            SceneRuntimeImpl sceneRuntime;
+
+            try { sceneRuntime = await sceneRuntimeFactory.CreateByPathAsync(sceneCodeUrl, instancePoolsProvider, sceneData.SceneShortInfo, ct, SceneRuntimeFactory.InstantiationBehavior.SwitchToThreadPool); }
+            catch (ScriptEngineException e)
+            {
+                // ScriptEngineException.ErrorDetails is ignored through the logging process which is vital in the reporting information
+                exceptionsHandler.OnJavaScriptException(new Exception(e.ErrorDetails));
+                throw;
```

Comment:
Yes, you are right. it is an existing vulnerability here. In theory everything that is passed to `SceneFacade` (similarly to its `Dispose`) should be disposed of. If you have time you can tackle it, otherwise it's not the big issue - we don't expect that our flow is drastically broken...